anxcloud/struct_kubernetes.go

```diff
@@ -0,0 +1,96 @@
+package anxcloud
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	kubernetesv1 "go.anx.io/go-anxcloud/pkg/apis/kubernetes/v1"
+	"go.anx.io/go-anxcloud/pkg/utils/pointer"
+)
+
+const gibiFactor = 1073741824 // math.Pow(2, 30)
+
+func unflattenNodePool(s map[string]interface{}) *kubernetesv1.NodePool {
+	return &kubernetesv1.NodePool{
+		Identifier: s["id"].(string),
+		Name:       s["name"].(string),
+		Replicas:   pointer.Int(s["replicas"].(int)),
+		CPUs:       s["cpus"].(int),
+		Memory:     s["memory"].(int) * gibiFactor,
```

Comment:
I've added automatic unit conversion so that the user doesn't have to type the size in bytes. WDYT?

Reply:
let's call then `memory_gib` and co and then it's ok